### DISCUSSION TITLE
Using LocalResponse::into_json in async tests

### DISCUSSION THREAD

**Discussion by bubelov** [rwf2/Rocket#1787] [Questions] [answered]
Hello again =) Are there any known issues with using `into_json` in tests? Haven't found any mentions in the doc and GitHub issues. Here are the two approaches I tried:

```rust
let req = client.get("/exchange_rates?base=USD&quote=EUR");
let res = req.dispatch().await;
assert_eq!(res.status(), Status::Ok);
let body = res.into_string().await.unwrap();
let body: ExchangeRate = json::from_str(&body).unwrap();
assert_eq!(rate, body);

let req = client.get("/exchange_rates?base=USD&quote=EUR");
let res = req.dispatch().await;
assert_eq!(res.status(), Status::Ok);
// Time to halt forever...
let body = res.into_json::<ExchangeRate>().await.unwrap();
assert_eq!(rate, body);
```

The first one has more boilerplate, so I tried to use `into_json` but it halts the test execution (deadlock?). If I switch to sync client then it simply fails with "trying to initialize a runtime inside runtime kind of error. Any ideas?

Here is the full project, calling `cargo test` simply hangs on this line:

https://github.com/bubelov/rocket-rates/blob/0f63685a82599b083a6b79ec79fda3a67f032a29/src/tests.rs#L37

**Comment by SergioBenitez** [comment, marked as answer]
Yes. The issue has been fixed in master for some time and will be in the next release.
